### PR TITLE
Is project onboarded endpoint

### PR DESCRIPTION
**Pull request by issaafalkattan**
Rough implementation of the isProjectOnboarded endpoint, needs cleaning up but works :yay

### CODE REVIEW

File: src/api/routes/project/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+// export { default } from './project';
```
Comment (frikkf) [inline src/api/routes/project/index.ts]: Do you want to include this?

File: src/api/routes/project/project.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Router from '@koa/router';
+import { TracedContext } from 'types';
+import validate from 'api/middleware/validate';
+import { APIError } from 'api/middleware/error';
+
+import isProjectOnboarded from '../../../services/projectServices';
+
+import { isProjectOnboardedSchema } from './validationSchema';
+
+export default (app: Router) => {
```
Comment (frikkf) [inline src/api/routes/project/project.ts]: I would rename app to router and then reuse the same router here. We cont even need to return the router.

File: src/api/routes/project/project.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const router = new Router();
+
+  router.get(
+    '/project',
+    validate({ query: isProjectOnboardedSchema }),
+    async (ctx: TracedContext) => {
```
Comment (frikkf) [inline src/api/routes/project/project.ts]: The route looks good :) Not really sure how we will handle APIErrors going forward, but we will learn that when we have to write more endpoints :)

File: src/api/routes/routes.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { APIError } from 'api/middleware/error';
 import validate from 'api/middleware/validate';
 import Joi from '@hapi/joi';
 
+import project from './project/project';
```
Comment (frikkf) [inline src/api/routes/routes.ts]: If you fix the export { default } file below, then you can import from ./project

File: src/db/db.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { Sequelize } from 'sequelize';
```
Comment (frikkf) [inline src/db/db.ts]: Since we are using another db library sequelize we can use something like this to make the query:

```
sequelize.query("UPDATE users SET y = 42 WHERE x = 12").then(([results, metadata]) => {
  // Results will be an empty array and metadata will contain the number of affected rows.
})
```

Im doing some research for the way we handle DB connections, so i think we can start using the sequelize.query the same way as we do with pg.

So we can replace the pg pool implementation with sequelize :)

File: src/db/db.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
...
 import config from 'config';
+import { Pool, QueryResult } from 'pg';
 
 const connect = async () => {
```
Comment (frikkf) [inline src/db/db.ts]: So to get the sequelize instance you have to split the connect function

```
export const database = new Sequelize(name, user, password, {
    host:
      environment === 'production'
        ? `/cloudsql/${connectionName}`
        : 'localhost',
    dialect: 'postgres',
  });

```

then you can just use database.query

File: src/types/index.ts [hunk 7]
```diff
@@ -1,9 +1,18 @@
...
 
 export type TracedOperation = { parentSpan: Span };
 
+export type TracedContext = ParameterizedContext<TracedState>;
+
 export type TracedState = { span: Span };
 
 export type RequestIdState = { requestId: string };
+
+export type BooleanPayload = boolean;
```
Comment (frikkf) [inline src/types/index.ts]: Why is this needed?

**Comment by frikkf** [comment]
Oh and I forgot. We can be a bit stricter with tests here aswell. So we should add some tests :) Let me know if you wonder about anything :)

File: src/types/index.ts (hunk 7 above)
Comment (issaafalkattan) [reply src/types/index.ts]: Its used in the dqm functions which is why i copied it but not sure if its useful anymore

Comment (frikkf) [reply src/types/index.ts]: ok :)
